tests/conftest.py

```diff
@@ -622,18 +624,39 @@ def xmlrpc(self, path, method, *args):
 
 
 @pytest.fixture
-def webtest(app_config):
-    # TODO: Ensure that we have per test isolation of the database level
-    #       changes. This probably involves flushing the database or something
-    #       between test cases to wipe any committed changes.
+def tm():
+    tm = transaction.TransactionManager(explicit=True)
+    tm.begin()
+    tm.doom()
 
+    yield tm
+
+    tm.abort()
+
+
+@pytest.fixture
+def webtest(app_config, tm, db_session):
     # We want to disable anything that relies on TLS here.
     app_config.add_settings(enforce_https=False)
 
-    try:
-        yield _TestApp(app_config.make_wsgi_app())
-    finally:
-        app_config.registry["sqlalchemy.engine"].dispose()
```

Comment:
Do you know why we not need to `dispose()` any more?

Reply:
In the `db_session` fixture, the call to `.dispose()` is already issued. I think the responsibility should be with that fixture.. not super sure. But it should only need to happen once.

Reply:
Thanks for clarifying!